src/ploomber_engine/ipython.py

```diff
@@ -152,6 +152,28 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_value, traceback):
         self.clear_instance()
 
+    def _get_interactive_variables(self):
+        """
+        similar to the method %who of,it returns
+        a sequence of all interactive variables
+        """
+        user_ns = self.user_ns
+        user_ns_hidden = self.user_ns_hidden
```

Comment:
user_ns_hidden is defined here :
https://github.com/ipython/ipython/blob/main/IPython/core/interactiveshell.py , at line 1192:
![image](https://user-images.githubusercontent.com/128071610/234835603-d31a2e84-667f-46d4-8b99-1532c300051c.png)

The aim is to use it with the "%who" command, and to filter out non interactve variiables
from the list user_ns (line 247)

![image](https://user-images.githubusercontent.com/128071610/234836353-99b35496-d93f-4e46-9c63-18abb1ecdd06.png)

I thought it would be more readable and interpretable to use a part of the core of the method directly rather than to call the associated magic method.

Reply:
got it, thanks for the explanation!